core/src/main/java/org/infinispan/scattered/impl/ScatteredStateConsumerImpl.java

```diff
@@ -105,6 +105,10 @@ protected void beforeTopologyInstalled(int topologyId, boolean startRebalance, C
       // and ScatteringInterceptor would check according to the new CH.
       for (int segment = 0; segment < newWriteCh.getNumSegments(); ++segment) {
          if (!newWriteCh.isSegmentLocalToNode(rpcManager.getAddress(), segment)) {
+            // We cancel any existing transfers as failed key transfer could move segment
+            // to OWNED state if it fails, concurrently with installing the topology.
```

Comment:
I'll remove the duplicate 'if fails'... KT does not install a topology but it can change segment state. Nothing prevents a new topology to come during KT, and topology changes segment state, too. Therefore in order to synchronize those two events we kill the former one.